pkg/executables/executables.go

```diff
@@ -61,47 +65,54 @@ func (e *linuxDockerExecutable) workingDirectory() (string, error) {
 
 func (e *linuxDockerExecutable) Execute(ctx context.Context, args ...string) (bytes.Buffer, error) {
 	var stdout bytes.Buffer
-	if command, err := e.buildCommand(map[string]string{}, e.cli, args...); err != nil {
+	if command, containerName, err := e.buildCommand(map[string]string{}, e.cli, args...); err != nil {
 		return stdout, err
 	} else {
+		defer executeCleanup(ctx, containerName)
 		return execute(ctx, "docker", nil, command...)
 	}
 }
 
 func (e *linuxDockerExecutable) ExecuteWithStdin(ctx context.Context, in []byte, args ...string) (bytes.Buffer, error) {
 	var stdout bytes.Buffer
-	if command, err := e.buildCommand(map[string]string{}, e.cli, args...); err != nil {
+	if command, containerName, err := e.buildCommand(map[string]string{}, e.cli, args...); err != nil {
 		return stdout, err
 	} else {
+		defer executeCleanup(ctx, containerName)
 		return execute(ctx, "docker", in, command...)
 	}
 }
 
-func (e *linuxDockerExecutable) buildCommand(envs map[string]string, cli string, args ...string) ([]string, error) {
+func (e *linuxDockerExecutable) buildCommand(envs map[string]string, cli string, args ...string) ([]string, string, error) {
 	directory, err := e.workingDirectory()
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	var envVars []string
 	for k, v := range envs {
 		envVars = append(envVars, "-e", fmt.Sprintf("%s=%s", k, v))
 	}
+	containerName := containerNamePrefix + cli + "_" + time.Now().Format("2006-01-0215.04.05.000000")
+
 	dockerCommands := []string{
-		"run", "-i", "--network", "host", "-v", fmt.Sprintf("%s:%s", directory, directory),
+		"run", "--name", containerName, "-i", "--network", "host", "-v", fmt.Sprintf("%s:%s", directory, directory),
 		"-w", directory, "-v", "/var/run/docker.sock:/var/run/docker.sock",
 	}
+
 	dockerCommands = append(dockerCommands, envVars...)
 	dockerCommands = append(dockerCommands, "--entrypoint", cli, e.image)
 	dockerCommands = append(dockerCommands, args...)
-	return dockerCommands, nil
+
+	return dockerCommands, containerName, nil
 }
 
 func (e *linuxDockerExecutable) ExecuteWithEnv(ctx context.Context, envs map[string]string, args ...string) (bytes.Buffer, error) {
 	var stdout bytes.Buffer
-	if command, err := e.buildCommand(envs, e.cli, args...); err != nil {
+	if command, containerName, err := e.buildCommand(envs, e.cli, args...); err != nil {
 		return stdout, err
 	} else {
+		defer executeCleanup(ctx, containerName)
```

Comment:
Do we want to error if cleaning up fails? If we do, we can probably just not use defer and call it below with error blocks. Wondering if we want to surface the error in cleaning up with at a lesser log level, something like 1 or even 0.

Reply:
I don't believe the cluster creation/deletion/updates should stop because we fail to clean up some docker containers or volumes. We could print a warning maybe that in case the container clean up fails multiple times, volumes can keep taking up more space and suggest the users to run something like `docker system prune -a`. We can bump the logging level for the clean ups to 1 or 0 if we want.